vspace/src/main/java/edu/asu/diging/vspace/web/staff/ExhibitionConfigurationController.java

```diff
@@ -41,11 +41,18 @@
     @RequestMapping("/staff/exhibit/config")
     public String showExhibitions(Model model) {
         // for now we assume there is just one exhibition
-        IExhibition exhibition = exhibitManager.getStartExhibition();
-        if(exhibition!=null) {
+        Exhibition exhibition = (Exhibition) exhibitManager.getStartExhibition();
+        if (exhibition != null) {
+            UUID randomUUID = UUID.randomUUID();
+            String randomString = randomUUID.toString().replaceAll("-", "");
+            exhibition.setPreviewId("EXH_PREVIEW_" + randomString.substring(0, 8));
```

Comment:
is this done every time the exhibition is configured? That doesn't make sense, it should only happen when there is no preview id yet.

Reply:
also, no hard-coding of strings. id prefix should be a constant and generating an id is business logic, so it should not be in a controller.